ui/app/actions.js

```diff
@@ -348,9 +348,25 @@ function navigateToNewAccountScreen () {
   }
 }
 
-function addNewAccount () {
+function addNewAccount (oldIdentities) {
   log.debug(`background.addNewAccount`)
-  return callBackgroundThenUpdate(background.addNewAccount)
+  return (dispatch) => {
+    dispatch(actions.showLoadingIndication())
+    return new Promise((resolve, reject) => {
+      background.addNewAccount((err, { identities: newIdentities}) => {
+        if (err) {
+          dispatch(actions.displayWarning(err.message))
+          return reject(err)
+        }
+        const newAccountAddress = Object.keys(newIdentities).find(address => !oldIdentities[address])
+
+        dispatch(actions.hideLoadingIndication())
+
+        forceUpdateMetamaskState(dispatch)
```

Comment:
why is this necessary?

Reply:
This is in the `callBackgroundThenUpdate` fubction. I replaced the `callBackgroundThenUpdate(ackground.addNewAccount)`  call with the code above, and didn't want to remove the `forceUpdateMetamaskState()` call.

I don't know exactly why it is necessary, but has to with maintaining the correct client side state when the background state changes.